test/cli/install/bun-yarn.test.ts

```diff
@@ -0,0 +1,236 @@
+  expect(await getPackageJSONContents()).toMatchSnapshot();
+})
+
+const yarn_lock_test_pkg_with_prerelease =
+`# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
+# yarn lockfile v1
+
+
+test_pkg_with_prerelease@^0.0.0-massively-long-prerelease-tag:
+  version "0.0.0-massively-long-prerelease-tag"
+  resolved "https://registry.yarnpkg.com/test_pkg_with_prerelease/-/test_pkg_with_prerelease-0.0.0-massively-long-prerelease-tag.tgz#1c35e8548bda8b57df0fa3a8e2d19adc2df871b7"
+  integrity sha512-XBCx9Ce+vAiBVsxLC6jAU7PPJuysLNZZhR+rT17IvVvGVm2MVsAffu0fmJwB1e7XDQ/9k2NM6ugFYpHPbzmuCA==
+`;
+
+// We test this separately because there is extra logic for prerelease version strings that are not inlineable by `String`
+// We do a little space optimization where the prerelease version literal can point inside the filename string.
+it("should work for files with long prerelease versions with version numbers in the filename", async () => {
+	const urls: string[] = [];
+  setHandler(dummyRegistry(urls, { "0.0.0-massively-long-prerelease-tag": {} }));
+  await makeBasicPackageJSON({ "test_pkg_with_prerelease": "^0.0.0-massively-long-prerelease-tag" });
+	await writeFile(join(package_dir, "yarn.lock"), yarn_lock_test_pkg_with_prerelease);
+  const { err, out, exited } = await command("install", "-y");
+  expect(await exited).toBe(0);
+  expect(err).toContain("Saved yarn.lock");
+  expect(out).toContain(`test_pkg_with_prerelease@0.0.0-massively-long-prerelease-tag`);
+  expect(await getYarnLockContents()).toMatchSnapshot();
+  expect(await getPackageJSONContents()).toMatchSnapshot();
+})
+
+const yarn_lock_test_pkg_with_prerelease2 =
+`# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
+# yarn lockfile v1
+# bun ./bun.lockb --hash: C4A807CA0191E97A-f12382d9a045f641-258DFF03787839E8-99738125a32e4587
+
+
+"test_pkg_with_prerelease_2@https://github.com/Validark/Validark/raw/master/test_pkg_with_prerelease_2_.tgz":
+  version "0.0.1-massively-long-prerelease-tag"
+  resolved "https://github.com/Validark/Validark/raw/master/test_pkg_with_prerelease_2_.tgz"
+`;
+
+it("should work for files with long prerelease versions with no version number in the filename", async () => {
+	const urls: string[] = [];
+  setHandler(dummyRegistry(urls, { "test_pkg_with_prerelease_2.tgz": {} }));
+  await makeBasicPackageJSON({ "test_pkg_with_prerelease_2": "https://github.com/Validark/Validark/raw/master/test_pkg_with_prerelease_2_.tgz" });
+	await writeFile(join(package_dir, "yarn.lock"), yarn_lock_test_pkg_with_prerelease2);
+  const { err, out, exited } = await command("install", "-y");
+  expect(await exited).toBe(0);
+  expect(err).toContain("Saved yarn.lock");
+  expect(out).toContain(`test_pkg_with_prerelease_2@https://github.com/Validark/Validark/raw/master/test_pkg_with_prerelease_2_.tgz`);
+  expect(await getYarnLockContents()).toMatchSnapshot();
+  expect(await getPackageJSONContents()).toMatchSnapshot();
+})
```

Comment:
To be honest I am not sure if this grabs the local copy or not. I think it doesn't.